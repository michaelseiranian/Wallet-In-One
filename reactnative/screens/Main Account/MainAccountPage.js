import { StatusBar } from 'expo-status-bar';
import {StyleSheet, Text, View, TouchableOpacity} from 'react-native';
import { useTheme } from 'reactnative/src/theme/ThemeProvider';
import { styles } from 'reactnative/screens/All_Styles.style.js';
import Icon from 'react-native-vector-icons/FontAwesome';
import Icon2 from 'react-native-vector-icons/Ionicons';

import React from 'react';


export default function MainAccountPage({ navigation }) {
    const { dark, colors, setScheme } = useTheme();

    const stylesInternal = StyleSheet.create({
        row: {
            flex: 1,
            flexDirection: 'row',
        },
        box: {
            flex: 1,
            borderRadius: 15,
            margin: 5,
            flexDirection: 'column',
            justifyContent: 'space-evenly',
            alignItems: 'center'
        },
        text: { 
            color: 'white', 
            fontSize: 23, 
            fontWeight: 'bold', 
            textAlign: 'center'
        }
    });

    return (
        <View style={[styles(dark, colors).container, {flexDirection: 'column', padding: 5,}]}>
            <TouchableOpacity
                style={[stylesInternal.box, { backgroundColor: '#5686f2' }]}
                onPress={() => navigation.navigate("Bank Accounts")}
            >
                <Icon style={{ color: 'white' }} name="bank" size={80} />
                <Text style={stylesInternal.text}>{'Bank Accounts'}</Text>
            </TouchableOpacity>
            <TouchableOpacity
                style={[stylesInternal.box, { backgroundColor: 'red' }]}
                onPress={() => navigation.navigate("Crypto Wallets & Exchanges")}
            >
                <Icon2 style={{ color: 'white' }} name="wallet" size={80} />
                <Text style={stylesInternal.text}>{'Cryptocurrency'}</Text>
            </TouchableOpacity>
            <TouchableOpacity
                style={[stylesInternal.box, { backgroundColor: '#55a755' }]}
                onPress={() => navigation.navigate("Stock Account List")}
            >
                <Icon style={{ color: 'white' }} name="line-chart" size={80} />
                <Text style={stylesInternal.text}>{'Stock Accounts'}</Text>
            </TouchableOpacity>
        </View>
    );
}
