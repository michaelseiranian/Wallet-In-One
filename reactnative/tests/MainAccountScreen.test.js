import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react-native';
import MainAccountPage from '../screens/Main Account/MainAccountPage';

describe('<MainAccountPage />', () => {
    it('mainAccountPage snapshot test', () => {
        const mainAccountPage = render(<MainAccountPage/>);
        expect(mainAccountPage).toMatchSnapshot();
    }
    )

    it('mainAccountPage banking button test', () => {
        const navigate = jest.fn();
        const mainAccountPage = render(<MainAccountPage navigation={ {navigate} }/>);
        fireEvent.press(screen.getByText('Bank Accounts'));
        expect(navigate).toHaveBeenCalledWith('Bank Accounts');
        expect(mainAccountPage).toMatchSnapshot();
    }
    )

    it('mainAccountPage cryptocurrency button test', () => {
        const navigate = jest.fn();
        const mainAccountPage = render(<MainAccountPage navigation={ {navigate} }/>);
        fireEvent.press(screen.getByText('Cryptocurrency'));
        expect(navigate).toHaveBeenCalledWith('Crypto Wallets & Exchanges');
        expect(mainAccountPage).toMatchSnapshot();
    }
    )

    it('mainAccountPage stocks button test', () => {
        const navigate = jest.fn();
        const mainAccountPage = render(<MainAccountPage navigation={ {navigate} }/>);
        fireEvent.press(screen.getByText('Stock Accounts'));
        expect(navigate).toHaveBeenCalledWith('Stock Account List');
        expect(mainAccountPage).toMatchSnapshot();
    }
    )
})