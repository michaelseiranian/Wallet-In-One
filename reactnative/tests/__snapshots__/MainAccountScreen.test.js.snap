// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainAccountPage /> mainAccountPage banking button test 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "white",
        "flex": 1,
      },
      {
        "flexDirection": "column",
        "padding": 5,
      },
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#5686f2",
        "borderRadius": 15,
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-evenly",
        "margin": 5,
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": undefined,
            "fontSize": 80,
          },
          {
            "color": "white",
          },
          {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      
    </Text>
    <Text
      style={
        {
          "color": "white",
          "fontSize": 23,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Bank Accounts
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "red",
        "borderRadius": 15,
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-evenly",
        "margin": 5,
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": undefined,
            "fontSize": 80,
          },
          {
            "color": "white",
          },
          {
            "fontFamily": "ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      
    </Text>
    <Text
      style={
        {
          "color": "white",
          "fontSize": 23,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Cryptocurrency
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#55a755",
        "borderRadius": 15,
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-evenly",
        "margin": 5,
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": undefined,
            "fontSize": 80,
          },
          {
            "color": "white",
          },
          {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      
    </Text>
    <Text
      style={
        {
          "color": "white",
          "fontSize": 23,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Stock Accounts
    </Text>
  </View>
</View>
`;

exports[`<MainAccountPage /> mainAccountPage cryptocurrency button test 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "white",
        "flex": 1,
      },
      {
        "flexDirection": "column",
        "padding": 5,
      },
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#5686f2",
        "borderRadius": 15,
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-evenly",
        "margin": 5,
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": undefined,
            "fontSize": 80,
          },
          {
            "color": "white",
          },
          {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      
    </Text>
    <Text
      style={
        {
          "color": "white",
          "fontSize": 23,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Bank Accounts
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "red",
        "borderRadius": 15,
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-evenly",
        "margin": 5,
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": undefined,
            "fontSize": 80,
          },
          {
            "color": "white",
          },
          {
            "fontFamily": "ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      
    </Text>
    <Text
      style={
        {
          "color": "white",
          "fontSize": 23,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Cryptocurrency
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#55a755",
        "borderRadius": 15,
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-evenly",
        "margin": 5,
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": undefined,
            "fontSize": 80,
          },
          {
            "color": "white",
          },
          {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      
    </Text>
    <Text
      style={
        {
          "color": "white",
          "fontSize": 23,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Stock Accounts
    </Text>
  </View>
</View>
`;

exports[`<MainAccountPage /> mainAccountPage snapshot test 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "white",
        "flex": 1,
      },
      {
        "flexDirection": "column",
        "padding": 5,
      },
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#5686f2",
        "borderRadius": 15,
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-evenly",
        "margin": 5,
        "opacity": 1,
      }
    }
  >
    <Text />
    <Text
      style={
        {
          "color": "white",
          "fontSize": 23,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Bank Accounts
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "red",
        "borderRadius": 15,
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-evenly",
        "margin": 5,
        "opacity": 1,
      }
    }
  >
    <Text />
    <Text
      style={
        {
          "color": "white",
          "fontSize": 23,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Cryptocurrency
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#55a755",
        "borderRadius": 15,
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-evenly",
        "margin": 5,
        "opacity": 1,
      }
    }
  >
    <Text />
    <Text
      style={
        {
          "color": "white",
          "fontSize": 23,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Stock Accounts
    </Text>
  </View>
</View>
`;

exports[`<MainAccountPage /> mainAccountPage stocks button test 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "white",
        "flex": 1,
      },
      {
        "flexDirection": "column",
        "padding": 5,
      },
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#5686f2",
        "borderRadius": 15,
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-evenly",
        "margin": 5,
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": undefined,
            "fontSize": 80,
          },
          {
            "color": "white",
          },
          {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      
    </Text>
    <Text
      style={
        {
          "color": "white",
          "fontSize": 23,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Bank Accounts
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "red",
        "borderRadius": 15,
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-evenly",
        "margin": 5,
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": undefined,
            "fontSize": 80,
          },
          {
            "color": "white",
          },
          {
            "fontFamily": "ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      
    </Text>
    <Text
      style={
        {
          "color": "white",
          "fontSize": 23,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Cryptocurrency
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#55a755",
        "borderRadius": 15,
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-evenly",
        "margin": 5,
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": undefined,
            "fontSize": 80,
          },
          {
            "color": "white",
          },
          {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      
    </Text>
    <Text
      style={
        {
          "color": "white",
          "fontSize": 23,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Stock Accounts
    </Text>
  </View>
</View>
`;
